### PR TITLE
SDK: Add supported Python versions as env variable

### PR DESCRIPTION
**Pull request by zrgt**
- Created a job to build python-matrix as shown in:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#example-using-an-output-to-define-two-matrices

- Used `python-matrix` for `sdk-test` & `sdk-static-analysis`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,59 +75,57 @@ jobs:
...
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[dev]
+      - name: Check typing with MyPy
+        run: |
+          mypy basyx test
+      - name: Check code style with PyCodestyle
+        run: |
+          pycodestyle --count --max-line-length 120 basyx test
```
Comment (s-heppner) [inline .github/workflows/ci.yml:128]: Could it be that the indentation broke here at some points?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -125,24 +134,24 @@ jobs:
...
+          pip install .[dev]
+      - name: Check typing with MyPy
+        run: |
+          mypy <(codeblocks python README.md)
+      - name: Check code style with PyCodestyle
+        run: |
+          codeblocks --wrap python README.md | pycodestyle --count --max-line-length 120 -
+      - name: Run readme codeblocks with Python
+        run: |
+          codeblocks python README.md | python
```
Comment (s-heppner) [inline .github/workflows/ci.yml:154]: Same here

**Review by s-heppner — COMMENTED** [review]
That's a really sleek way of checking the versions! Thanks for the suggestion.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (zrgt) [reply .github/workflows/ci.yml:128]: No, this was an automatic linting of PyCharm

File: .github/workflows/ci.yml (hunk 2 above)
Comment (zrgt) [reply .github/workflows/ci.yml:154]: No, this was an automatic linting of PyCharm

**Review by s-heppner — APPROVED** [review]

**Pull request merged**